### DISCUSSION TITLE
How should I focus an AppWindow?

### DISCUSSION THREAD

**Discussion by JohnCido** [microsoft/WindowsAppSDK#1746] [Q&A] [answered]
How should I let AppWindow get the focus and elevate to the top from other windows?

**Comment by martibravo** [comment]
If you want to focus it, 
```cs 
yourAppWindow.Show(); //or yourAppWindow.Show(true);
``` 
Unfocus with 
```cs
yourAppWindow.Show(false);
```
Want to always stay on top? 

If it's a normal window (OverlappedPresenter):
```cs
var presenter = yourAppWindow.Presenter as OverlappedPresenter;
presenter.IsAlwaysOnTop = true;
```

**Reply by JohnCido** [reply]
According to the answer from @andrewleader and my experience trying the `Show()` method in my codes, this does not work at this moment.

**Reply by martibravo** [reply]
You are absolutely right, I apologize. I got it wrong.

**Comment by andrewleader** [comment, marked as answer]
Hi @JohnCido, the AppWindow APIs don't support bringing the window to the foreground (I assume you are trying to bring your window to the foreground/front as a result of an app launch redirection or some other event that should bring your window to the foreground). The `Show()` API unfortunately doesn't do that.

Here's how to bring a window to the foreground from a C# desktop app...

### 1. Install Microsoft.Windows.CsWin32 NuGet package
### 2. Add NativeMethods.txt file

This file specifies what C++ Win32 APIs you want projected into C#. We only need two APIs, so the file contents should simply be...

```
SetForegroundWindow
ShowWindow
```

### 3. Bring your window to the foreground

```csharp
    // Bring the window to the foreground... first get the window handle...
    IntPtr windowHandle = WinRT.Interop.WindowNative.GetWindowHandle(m_window);

    // Restore window if minimized... requires Microsoft.Windows.CsWin32 NuGet package and a NativeMethods.txt file with ShowWindow method
    Windows.Win32.PInvoke.ShowWindow((Windows.Win32.Foundation.HWND)windowHandle, Windows.Win32.UI.WindowsAndMessaging.SHOW_WINDOW_CMD.SW_RESTORE);

    // And call SetForegroundWindow... requires Microsoft.Windows.CsWin32 NuGet package and a NativeMethods.txt file with SetForegroundWindow method
    Windows.Win32.PInvoke.SetForegroundWindow((Windows.Win32.Foundation.HWND)windowHandle);
```

**Reply by JohnCido** [reply]
Thanks for the answer, it works now. What I want to know is do you have any plan to add it to the SDK using normal modern C# codes? I believe it's an important function to many desktop apps.

**Reply by andrewleader** [reply]
Yes, making this a straightforward API call is feedback we've shared with the AppWindow team, I'll record your request for that API! We don't have any plans to share yet about if/when that API would come though.
